pose_evaluation/metrics/test_embedding_distance_metric.py

```diff
@@ -0,0 +1,409 @@
+    hyp = random_tensor(768)
+    ref = random_tensor(769)
+
+    with pytest.raises(RuntimeError):
+        # gives RuntimeError: mat1 and mat2 shapes cannot be multiplied (1x768 and 769x1
+        # TODO: we should probably raise a more descriptive/helpful error/ ValueError
+        call_and_call_with_inputs_swapped(hyp, ref, cosine_metric.score)
+
+
+def test_invalid_input_single_number(cosine_metric: EmbeddingDistanceMetric) -> None:
+    hyp = random_tensor(768)
+    for ref in range(-2, 2):
+        with pytest.raises(IndexError):
+            # TODO: we should probably raise a more descriptive/helpful error/ ValueError
+            # IndexError: Dimension out of range (expected to be in range of [-1, 0], but got 1)
+            call_and_call_with_inputs_swapped(hyp, ref, cosine_metric.score)
+
+    logger.info("Invalid input successfully crashed as expected.")
+
+
+def test_invalid_input_noncontainernonnumber_types(cosine_metric: EmbeddingDistanceMetric) -> None:
+    hyp = random_tensor(768)
+    invalid_inputs = ["invalid_input", True]
+    for ref in invalid_inputs:
+        with pytest.raises((TypeError, IndexError)):
+            # TypeError: new(): invalid data type 'str'
+            # but True gives IndexError
+            # TODO: better TypeError, more descriptive
+            call_and_call_with_inputs_swapped(hyp, ref, cosine_metric.score)
+
+
+def test_invalid_input_empty_containers(cosine_metric: EmbeddingDistanceMetric) -> None:
+    """Test the metric with invalid inputs."""
+    emb1 = random_tensor(768)
+    invalid_inputs = ["", [], {}, tuple(), set()]
+
+    for invalid_input in invalid_inputs:
+        with pytest.raises((RuntimeError, TypeError, IndexError)):
+            # gives RuntimeError: mat1 and mat2 shapes cannot be multiplied (1x768 and 0x1)
+            # "" gives TypeError: new(): invalid data type 'str'
+            # IndexError: Dimension out of range (expected to be in range of [-1, 0], but got 1)
+            # TODO: we should probably raise a more descriptive/helpful error/ ValueError
+            call_and_call_with_inputs_swapped(emb1, invalid_input, cosine_metric.score)
+
+    logger.info("Invalid input successfully crashed as expected.")
+
+
+def test_score_tensor_input(cosine_metric):
+    """Test score function with torch.Tensor inputs."""
+    emb1 = torch.rand(768)
+    emb2 = torch.rand(768)
+
+    score = cosine_metric.score(emb1, emb2)
+    assert isinstance(score, float), "Output should be a float."
+
+
+def test_score_ndarray_input(cosine_metric):
+    """Test score function with np.ndarray inputs."""
+    emb1 = np.random.rand(768)
+    emb2 = np.random.rand(768)
+
+    score = cosine_metric.score(emb1, emb2)
+    assert isinstance(score, float), "Output should be a float."
+
+
+def test_score_all_list_of_lists_of_floats(cosine_metric):
+    """Does a 2D list of floats work?"""
+    hyps = [[np.random.rand() for _ in range(768)] for _ in range(5)]
```

Comment:
just to avoid np in this test, maybe `random.uniform(-1,1)` instead?

Reply:
Well, I use it in a few places, e.g. for np.pi or np.savetext